File: perl/words/lib/numerical.pl
#!/usr/bin/perl
use strict;
use warnings;

sub fibonacci {
	my ($n) = @_;
  if ($n < 0) {
    return -1;
  } elsif ($n < 2) {
    return $n;
  }
  my $acc1 = 1;
  my $acc2 = 0;
  my $temp;
  while ($n > 2) {
		$temp = $acc1;
		$acc1 = $acc1 + $acc2;
		$acc2 = $temp;
		$n--;
	}
	return $acc1 + $acc2;
}

sub factorial {
	my ($n) = @_;
  if ($n < 0) {
    return -1;
  }
	my $acc = 1;
	while ($n > 1) {
		$acc = $acc * $n;
		$n--;
	}
	return $acc;
}

sub gcd {
	my ($a, $b) = @_;
	while ($b > 0) {
		my $temp = $b;
		$b = $a % $b;
		$a = $temp;
	}
	return $a;
}

sub lcm {
	my ($a, $b) = @_;
	return ($a / gcd($a, $b)) * $b;
}

1;

File: dart/words/lib/numerical.dart
int fibonacci(int n) {
  if(n < 0) {
    return -1;
  } else if(n < 2) {
    return n;
  }
  int acc1 = 1;
  int acc2 = 0;
  while(n > 2) {
    [acc2, acc1] = [acc1, acc1 + acc2];
    n--;
  }
  return acc1 + acc2;
}

int factorial(int n) {
  if(n < 0) {
    return -1;
  }
  int acc = 1;
  while(n > 1) {
    acc *= n--;
  }
  return acc;
}

int gcd(int a, int b) {
  while(b != 0) {
    [a, b] = [b, a % b];
  }
  return a;
}

int lcm(int a, int b) {
  return (a ~/ gcd(a, b)) * b;
}